clippy_lints/src/as_conversions.rs

```diff
@@ -0,0 +1,45 @@
+use rustc::lint::{in_external_macro, EarlyContext, EarlyLintPass, LintArray, LintContext, LintPass};
+use rustc::{declare_lint_pass, declare_tool_lint};
+use syntax::ast::*;
+
+use crate::utils::span_help_and_lint;
+
+declare_clippy_lint! {
+    /// **What it does:** Checks for usage of `as` conversions.
+    ///
+    /// **Why is this bad?** `as` conversions will perform many kinds of
+    /// conversions, including silently lossy conversions and dangerous coercions.
+    /// There are cases when it's necessary to use `as`, so the lint is
+    /// Allow by default.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    /// ```rust
+    /// let a: u32 = 0;
+    /// let p = &a as *const u32 as *mut u32;
```

Comment:
Could you add an example that shows the better alternative here? Similar to https://rust-lang.github.io/rust-clippy/master/index.html#default_trait_access